core/src/main/scala/org/apache/spark/ui/jobs/UIData.scala

```diff
@@ -127,6 +127,14 @@ private[spark] object UIData {
     def updateTaskMetrics(metrics: Option[TaskMetrics]): Unit = {
       _metrics = TaskUIData.toTaskMetricsUIData(metrics)
     }
+
+    def taskDuration: Long = {
+      if (taskInfo.status == "RUNNING") {
+        _taskInfo.timeRunning(System.currentTimeMillis)
+      } else {
+        _metrics.map(_.executorRunTime).getOrElse(1L)
```

Comment:
`executorRunTime` will not get set in some failure handling scenarios:
https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/executor/Executor.scala#L403

hence the option.  I don't think it make sense to return `1L` in those cases in the rest api -- it should probably keep it as an option.  (In the current code, the UI will use `1L` just for sorting, but displays an empty string -- that seems more consistent with keeping it as an option.)